### PR TITLE
[PS-875] sdk bug fix on uid

### PR DESCRIPTION
**Pull request by swVisenze**

### CODE REVIEW

**Comment by swVisenze** [comment]
1 fix bug on uid, avoid generate every time 
2. add update privacy policy in sdk

File: productcat/src/main/java/com/visenze/productcat/android/UpdatePrivacyParams.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    public Map<String, List<String>> toMap() {
+        Map<String, List<String> > map = super.toMap();
+
+        putStringInMap(map, OPT_IN, Boolean.toString(optIn));
+
+        if(email != null) {
+            putStringInMap(map, EMAIL, email);
+        }
+
+        putStringInMap(map, VERSION, "1.0.0");
```
Comment (thehung111) [inline productcat/src/main/java/com/visenze/productcat/android/UpdatePrivacyParams.java:31]: this should not be hardcoded but meant to be read from a setting e.g. for version 2.x, we have privacy version 1.0.0, for version 3.x, we have a newer privacy version. The behavior is to first retrieved the current privacy setting version, then pass to server.

But we can do this later.

File: productcat/src/main/java/com/visenze/productcat/android/ProductCat.java [hunk 2]
```diff
@@ -152,6 +152,14 @@ public boolean isPrivacyShowed() {
         return mPrivacyPolicy.isPrivacyShowed();
     }
 
+    public boolean isAdsAccepted() {
+        return mPrivacyPolicy.isAdsAccepted();
+    }
+
+//    public void setAdsAceepted(boolean accepted) {
```
Comment (thehung111) [inline productcat/src/main/java/com/visenze/productcat/android/ProductCat.java:159]: can remove these

Comment (swVisenze) [reply productcat/src/main/java/com/visenze/productcat/android/ProductCat.java]: removed and pushed to 2.1.5

**Pull request merged**